chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/DeveloperManagerImpl.java

```diff
@@ -123,6 +142,14 @@ public DeveloperDTO getById(final Long id) throws EntityRetrievalException {
                 developer.getTransparencyAttestationMappings().add(map);
             }
         }
+
+        //Remove any deleted status events
```

Comment:
Why are deleted status events getting returned that then need to be removed? Should the dao be changed to not return the deleted status events?

Reply:
You are 100% correct, except for some reason it was not working working in certain situations for me.
If the method is called before deleting any status events, the deleted status events in the DB are not returned and it worked as expected.
If the method is called after deleting status events, the deleted status events are returned.  They are marked as deleted, but they are part of the resultset.

I think this may have something to do with hibernate caching or some sort of txn issue.

Reply:
I was thinking in the DeveloperDAOImpl getById method the HQL is joining the status events but not excluding those marked as deleted. Could you try changing that HQL and see if you don't need the lines removing deleted events in this manager?

Reply:
I've tried.  There is also Where annotation in the DeveloperEntity:
```
 @OneToMany(fetch = FetchType.LAZY, mappedBy = "developerId")
    @Basic(optional = false)
    @Column(name = "vendor_id", nullable = false)
    @Where(clause = "deleted <> 'true'")
    private Set<DeveloperStatusEventEntity> statusEvents = new LinkedHashSet<DeveloperStatusEventEntity>();
```

I am out of ideas, that's I had to go with the hacky solution.

Reply:
Can you look into this a little more? Locally, I went to Manage Developers and Products, selected Acmeware, Inc. which only had a single status event of "Active". I added a status event of Suspended by ONC with a date of 10/24/2018 and reason "test" and saved it. I could see the new status event in the database and also in the UI and the JSON results in Chrome's developer tools. I then edited that developer again and deleted the Suspended by ONC status. I could see in the database it was marked as deleted. It did not come back in the JSON results any longer so I'm wondering how you were seeing that happen or if it could be a result of some other change. In addition I turned on printing of all the generated Hibernate SQL (a log4j setting plus two lines changed in persistence.xml - not sure if all those changes are required but at least some of them are) and I do see it filtering out deleted statuses like you would expect from the annotations you showed above. I'm just worried a change has been introduced requiring the hacky solution and it would be better to take a little more time and figure it out vs proceeding as-is.
`   select
        distinct developere0_.vendor_id as vendor1_52_0_,
        addressent1_.address_id as address1_135_1_,
        contactent2_.contact_id as contact1_1_2_,
        statuseven3_.vendor_status_history_id as vendor1_63_3_,
        developers4_.vendor_status_id as vendor1_94_4_,
        developerc5_.vendor_id as vendor1_69_5_,
        developere0_.address_id as address10_52_0_,
        developere0_.contact_id as contact2_52_0_,
        developere0_.creation_date as creation3_52_0_,
        developere0_.deleted as deleted52_0_,
        developere0_.vendor_code as vendor5_52_0_,
        developere0_.last_modified_date as last6_52_0_,
        developere0_.last_modified_user as last7_52_0_,
        developere0_.name as name52_0_,
        developere0_.website as website52_0_,
        addressent1_.city as city135_1_,
        addressent1_.country as country135_1_,
        addressent1_.creation_date as creation4_135_1_,
        addressent1_.deleted as deleted135_1_,
        addressent1_.last_modified_date as last6_135_1_,
        addressent1_.last_modified_user as last7_135_1_,
        addressent1_.state as state135_1_,
        addressent1_.street_line_1 as street9_135_1_,
        addressent1_.street_line_2 as street10_135_1_,
        addressent1_.zipcode as zipcode135_1_,
        contactent2_.creation_date as creation11_1_2_,
        contactent2_.deleted as deleted1_2_,
        contactent2_.email as email1_2_,
        contactent2_.friendly_name as friendly4_1_2_,
        contactent2_.full_name as full5_1_2_,
        contactent2_.last_modified_date as last6_1_2_,
        contactent2_.last_modified_user as last7_1_2_,
        contactent2_.phone_number as phone8_1_2_,
        contactent2_.signature_date as signature9_1_2_,
        contactent2_.title as title1_2_,
        statuseven3_.creation_date as creation2_63_3_,
        statuseven3_.deleted as deleted63_3_,
        statuseven3_.vendor_id as vendor4_63_3_,
        statuseven3_.vendor_status_id as vendor5_63_3_,
        statuseven3_.last_modified_date as last6_63_3_,
        statuseven3_.last_modified_user as last7_63_3_,
        statuseven3_.reason as reason63_3_,
        statuseven3_.status_date as status9_63_3_,
        statuseven3_.vendor_id as vendor4_52_0__,
        statuseven3_.vendor_status_history_id as vendor1_0__,
        developers4_.creation_date as creation2_94_4_,
        developers4_.deleted as deleted94_4_,
        developers4_.last_modified_date as last4_94_4_,
        developers4_.last_modified_user as last5_94_4_,
        developers4_.name as name94_4_,
        developerc5_.active as active69_5_,
        developerc5_.retired as retired69_5_,
        developerc5_.suspended_by_acb as suspended4_69_5_,
        developerc5_.suspended_by_onc as suspended5_69_5_,
        developerc5_.terminated_by_onc as terminated6_69_5_,
        developerc5_.withdrawn_by_acb as withdrawn7_69_5_,
        developerc5_.withdrawn_by_developer as withdrawn8_69_5_ 
    from
        vendor developere0_ 
    left outer join
        address addressent1_ 
            on developere0_.address_id=addressent1_.address_id 
    left outer join
        contact contactent2_ 
            on developere0_.contact_id=contactent2_.contact_id 
    **left outer join
        vendor_status_history statuseven3_ 
            on developere0_.vendor_id=statuseven3_.vendor_id 
            and (
                statuseven3_.deleted <> 'true'
            )** 
    left outer join
        vendor_status developers4_ 
            on statuseven3_.vendor_status_id=developers4_.vendor_status_id 
    left outer join
        developer_certification_statuses developerc5_ 
            on developere0_.vendor_id=developerc5_.vendor_id 
    where
        developere0_.deleted<>true`